grails-app/services/org/pih/warehouse/putaway/PutawayService.groovy

```diff
@@ -38,37 +40,25 @@ class PutawayService {
     GrailsApplication grailsApplication
 
     def getPutawayCandidates(Location location) {
-        List putawayItems = []
-        List<Location> internalLocations = locationService.getInternalLocations(location,
-                [ActivityCode.RECEIVE_STOCK] as ActivityCode[])
-
         List binLocationEntries = productAvailabilityService.getAvailableQuantityOnHandByBinLocation(location)
 
-        internalLocations.each { internalLocation ->
-            List putawayItemsTemp = binLocationEntries.findAll {
-                it.binLocation == internalLocation
-            }
-            if (putawayItemsTemp) {
-                putawayItemsTemp = putawayItemsTemp.collect {
-
-                    // FIXME Removed because it was causing very slow performance - determine if this is even necessary
-                    List<AvailableItem> availableItems = []
-
-                    PutawayItem putawayItem = new PutawayItem()
-                    putawayItem.putawayStatus = PutawayStatus.READY
-                    putawayItem.product = it.product
-                    putawayItem.inventoryItem = it.inventoryItem
-                    putawayItem.currentFacility = location
-                    putawayItem.currentLocation = it.binLocation
-                    putawayItem.putawayFacility = null
-                    putawayItem.putawayLocation = null
-                    putawayItem.availableItems = availableItems
-                    putawayItem.quantity = it.quantity
-                    return putawayItem
-                }
-                putawayItems.addAll(putawayItemsTemp)
+        List<PutawayItem> putawayItems = binLocationEntries.inject ([], { putawayItems,  binLocationEntry ->
+            if (binLocationEntry.binLocation.supports(ActivityCode.RECEIVE_STOCK)) {
+                return putawayItems + new PutawayItem(
+                        putawayStatus: PutawayStatus.READY,
+                        product: binLocationEntry.product,
+                        inventoryItem: binLocationEntry.inventoryItem,
+                        currentLocation: binLocationEntry.binLocation,
+                        currentFacility: location,
+                        putawayFacility: null,
+                        putawayLocation: null,
+                        availableItems: [],
```

Comment:
Maybe we should add the comment that was originally here about removing the available items?